src/devices/Vl53L1X/Vl53L1X.cs

```diff
@@ -0,0 +1,959 @@
+                    default:
+                        throw new Exception($"The returned range status code {rgSt} of the device is unknown.");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the offset correction value in mm.
+        /// </summary>
+        public short Offset
+        {
+            get
+            {
+                short offset = ReadInt16((ushort)Registers.ALGO__PART_TO_PART_RANGE_OFFSET_MM);
+                offset <<= 3;
+                offset /= 32;
+                return offset;
+            }
+            set
+            {
+                WriteInt16((ushort)Registers.ALGO__PART_TO_PART_RANGE_OFFSET_MM, (short)(value * 4));
+                WriteInt16((ushort)Registers.MM_CONFIG__INNER_OFFSET_MM, 0x0);
+                WriteInt16((ushort)Registers.MM_CONFIG__OUTER_OFFSET_MM, 0x0);
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the crosstalk correction value in cps.
+        /// This is the number of photons reflected back from the cover glass in cps.
+        /// </summary>
+        public ushort Xtalk
+        {
+            get
+            {
+                ushort xTalk = ReadUInt16((ushort)Registers.ALGO__CROSSTALK_COMPENSATION_PLANE_OFFSET_KCPS);
+
+                return (ushort)((xTalk * 1000) >> 9); /* * 1000 to convert kcps to cps and >> 9 (7.9 format) */
+            }
+            set
+            {
+                WriteUInt16((ushort)Registers.ALGO__CROSSTALK_COMPENSATION_X_PLANE_GRADIENT_KCPS, 0x0000);
+                WriteUInt16((ushort)Registers.ALGO__CROSSTALK_COMPENSATION_Y_PLANE_GRADIENT_KCPS, 0x0000);
+                WriteUInt16((ushort)Registers.ALGO__CROSSTALK_COMPENSATION_PLANE_OFFSET_KCPS,
+                    (ushort)((value << 9) / 1000)); /* * << 9 (7.9 format) and /1000 to convert cps to kpcs */
+            }
+        }
+
+        /// <summary>
+        /// This function programs the threshold detection mode.
+        /// For example:
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.Below): below 100
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.Above): above 300
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.Out): out-of-window
+        /// SetDistanceThreshold(dev,100,300, WindowDetectionMode.In): in window
+        /// </summary>
+        /// <param name="threshLow">
+        /// The threshold under which the device raises an interrupt if detectionMode =
+        /// WindowDetectionMode.Below
+        /// </param>
+        /// <param name="threshHigh">
+        /// The threshold above which the device raises an interrupt if detectionMode =
+        /// WindowDetectionMode.Above
+        /// </param>
+        /// <param name="detectionMode">The <see cref="WindowDetectionMode" /> where 0 = below, 1 = above, 2 = out, and 3 = in</param>
+        public void SetDistanceThreshold(ushort threshLow, ushort threshHigh, WindowDetectionMode detectionMode)
```

Comment:
if the thresholds are distances as well, then makes sense to use UnitsNet as well

Reply:
Now using Length instead of ushort